R/reactives.R

```diff
@@ -1838,7 +1817,8 @@ reactivePoll <- function(intervalMillis, session, checkFunc, valueFunc) {
     # firing and hold onto resources.
     if (re_finalized) {
       o$destroy()
-      rm(o, envir = parent.env(environment()))
+      # Note that rlang::as_function() inserts an additional parent environment.
+      rm(o, envir = parent.env(parent.env(environment())))
```

Comment:
what is this removing `o` from? The `reactivePoll()` function environment?

Why not capture the environment value before 1814 and pass it in at 1821?

Reply:
It does remove `o` from the `reactivePoll()` function environment.

Storing the environment vs. calling `parent.env(parent.env(environment()))`... Seems like six of one, half a dozen of the other.

Reply:
Storing the environment variable is safe and would allow rlang to change their implementation.

Your proposed change above is no different than the previous code, but now we depend on an outside function to create the environments.

Reply:
Fair enough, I've changed it to store the environment.